### DISCUSSION TITLE
Generate typescript for render variables

### DISCUSSION THREAD

**Discussion by maximelebastard** [harttle/liquidjs#539] [Q&A] [answered]
Hi, I'd like to make my templating safer using typescript.
My goal is to statically detect missing or wrongly typed variables.

That tool would generate types from this:
```liquid
Hello {{name}}
{{ if age is not empty and age > 10 }}
You are {{ age }} years old
{{ endif }}
```

```ts
type TemplateVars {
  name: string;
  age?: number;
}
```

Is there any kind of method or tool to do that automatically ?

**Comment by harttle** [comment, marked as answer]
We don't have this feature currently. And it's not easy to implement without understanding every block tag. There's more info here: #528 #421
